ft(registration-client-182201524):Users should be able Signup/Registe…

kalisaNkevin opened the pull request:
#### What does this PR do?

Users should be able Signup/Register to Barefoot Nomad

#### Description of Task to be completed?

`.`The sign up form should present to the user descriptive validation errors incase of any.
`.`The user should receive verification link on email upon successful registration

#### How should this be manually tested?

After cloning this repository, Switch to this branch:  `ft-registration-client-182201524`
The run: `yarn install `
To run the app:  `yarn start`
To run the test:  `yarn test`

#### Any background context you want to provide?

Fetching API from the backend and material UI framework for css 

#### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/182201524

munyanezaarmel reviewed (commented):
You don't have to import axios again it is already set you have to import API which found in utils and also backend UR should be in environment variables

kalisaNkevin commented:
> You don't have to import axios again it is already set you have to import API which found in utils and also backend UR should be in environment variables

munyanezaarmel reviewed (commented):
Works fine but I can suggest when registration is successful a page should navigate to login screen instead of home screen also check coveralls

ericus123 commented:
<img width="660" alt="Screenshot 2022-08-14 at 18 01 31" src="https://user-images.githubusercontent.com/58434968/184545233-db351775-394a-442f-bc26-769a1109dae9.png">

Fix the validations.
- I'm seeing errors for no reason.
- These errors are confusing.
`enter your username is equired`
either say `enter your username` or `username is equired`.
do same for other inputs as well

ericus123 commented on test/app.test.js:
```diff
@@ -2,7 +2,12 @@ import React from 'react'
 test('Initial render, Welcome to BareFoot Nomad frontEnd Tests', () => {
   render(<App />)
 })
+test('It should test all Sign up lines', () => {
+  render(<Signup />)
+})
+
```
add a signup test file separately and add signup tests.
test the inputs individually as well.

ericus123 commented on src/Views/RegisterView.js:
```diff
@@ -0,0 +1,11 @@
+
+function RegisterView() {
+  return (
+    <div>
```
remove these div tags and have signup component only here

munyanezaarmel committed:
fix(axios):fixing issues with webpack dotenv (#29)

fix(axios):fixing issues with webpack dotenv

ft(registration-client-182201524):Users should be able Signup/Register to Barefoot Nomad

ready for review

done

wor

woyu

done

last

op

changes

final

work

ok

ok

work

(ft-register-user):sign up on barefoot Nomad

done

(ft-sign-user):register sign up

ft(user-register):sign up

(ft-user-registration): user sign up

ft(sign-client):user registration

ft(user-registration):User sign up to barefoot Nomad

last commit

work done

done

done

(ft-register-account):sign up

(ft-user-signin-): registration BN

ericus123 commented on test/registartion.test.js:
```diff
@@ -0,0 +1,7 @@
+
+test('It should test all Sign up lines', () => {
+  render(<Signup />)
+})
```
is this the only test for registration ???

ericus123 commented:
Signup doesn't work???
![Uploading Screenshot 2022-08-15 at 07.59.36.png…]()

ericus123 commented:
Make sure to use only material UI alerts

ericus123 commented:
@kalisaNkevin  How is `dist` folder visible among changes?
Isn't it ignored??
Make sure it doesn't appear among git changes.

kalisaNkevin committed:
(ft-signin-user):register user

munyanezaarmel reviewed (approved):
add .dist in gitignore

kalisaNkevin committed:
(ft-user-reg):signin
